File: solutions/1503. Last Moment Before All Ants Fall Out of a Plank/1503.js
// Time complexity: O(n + m)
// Space complexity: O(1)
var getLastMoment = function(n, left, right) {
    let ans = 0;
    
    for (let num of left) {
        ans = Math.max(ans, num);
    }
    
    for (let num of right) {
        ans = Math.max(ans, n - num);
    }
    
    return ans;
};

File: solutions/1503. Last Moment Before All Ants Fall Out of a Plank/1503.cpp
// Time complexity: O(n + m)
// Space complexity: O(1)
class Solution {
public:
    int getLastMoment(int n, const vector<int>& left, const vector<int>& right) {
        int ans = 0;
        for (int num : left) {
            ans = max(ans, num);
        }
        
        for (int num : right) {
            ans = max(ans, n - num);
        }
        
        return ans;
    }
};